src/main/java/com/example/onlinebookstore/service/impl/CartItemServiceImpl.java

```diff
@@ -0,0 +1,62 @@
+package com.example.onlinebookstore.service.impl;
+
+import com.example.onlinebookstore.dto.CartItemRequestDto;
+import com.example.onlinebookstore.dto.CartItemResponseDto;
+import com.example.onlinebookstore.exception.EntityNotFoundException;
+import com.example.onlinebookstore.mapper.CartItemMapper;
+import com.example.onlinebookstore.model.Book;
+import com.example.onlinebookstore.model.CartItem;
+import com.example.onlinebookstore.model.ShoppingCart;
+import com.example.onlinebookstore.model.User;
+import com.example.onlinebookstore.repository.BookRepository;
+import com.example.onlinebookstore.repository.CartItemRepository;
+import com.example.onlinebookstore.service.CartItemService;
+import com.example.onlinebookstore.service.ShoppingCartService;
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CartItemServiceImpl implements CartItemService {
+    private final ShoppingCartService shoppingCartService;
+    private final CartItemMapper cartItemMapper;
+    private final CartItemRepository cartItemRepository;
+    private final BookRepository bookRepository;
+
+    @Override
+    @Transactional
+    public CartItemResponseDto add(CartItemRequestDto requestDto, User user) {
+        ShoppingCart shoppingCart = shoppingCartService.findByUserId(user);
+        CartItem cartItem = cartItemMapper.toEntity(requestDto);
+        Book book = bookRepository.findById(requestDto.getBookId())
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Book not found with id: " + requestDto.getBookId()));
+        cartItem.setBook(book);
+        cartItem.setShoppingCart(shoppingCart);
+        cartItemRepository.save(cartItem);
+        return cartItemMapper.toDto(cartItem);
+    }
+
+    @Override
+    @Transactional
+    public CartItemResponseDto updateCartItemQuantity(
+            Long id,
+            CartItemRequestDto requestDto,
+            User user
+    ) {
+        CartItem cartItem = cartItemRepository.findCartItemByBookId(requestDto.getBookId());
+        cartItem.setQuantity(requestDto.getQuantity());
+        cartItem = cartItemRepository.save(cartItem);
+        return cartItemMapper.toDto(cartItem);
+    }
+
+    @Override
+    @Transactional
+    public void delete(CartItemRequestDto requestDto) {
+        if (cartItemRepository.existsById(requestDto.getBookId())) {
+            cartItemRepository.delete(cartItemRepository
+                    .findCartItemByBookId(requestDto.getBookId()));
+        }
```

Comment:
should we just ignore it if it does not exist? Also let's not put the main flow and logic inside the if body, and stick to the [return early pattern](https://medium.com/swlh/return-early-pattern-3d18a41bba8)

Reply:
Change method